src/basic/smack-util.c

```diff
@@ -65,6 +134,21 @@ int mac_smack_apply(const char *path, const char *label) {
         return r;
 }
 
+int mac_smack_read_fd(int fd, const char **label) {
+        int r = 0;
+
```

Comment:
`assert(path);`

And in all cases, I think you should as well

`assert(label);`